instrumentation/rack/lib/opentelemetry/instrumentation/rack/middlewares/tracer_middleware.rb

```diff
@@ -93,7 +93,9 @@ def call(env) # rubocop:disable Metrics/MethodLength
           private
 
           def untraced_request?(env)
-            return true if @untraced_endpoints.include?(env['PATH_INFO'])
+            return true if @untraced_endpoints.any? do |endpoint|
+              endpoint === env['PATH_INFO'] # rubocop:disable Style/CaseEquality
+            end
```

Comment:
You've allowed a _lot_ more than just regular expressions this way. As noted [in this post](https://medium.com/rubyinside/triple-equals-black-magic-d934936a6379), triple-equals will test for a surprising range of truthiness.

For example, this would now allow us to pass:
- regular expressions (as intended)
- ranges
- classes
- procs
- ip addresses

...and we didn't want all of that. We should explicitly test for regular expressions if that's what we want to include, or the PR should state that it's allowing something much broader so we can figure out if we want that behavior or not.

Reply:
`env['PATH_INFO']` will always be a String. Using `===` on any of those will return false, with the lone exception of the `String` class (`String === "foo" #=> true`)

Anyway, I can change it if you think it's an issue.

Reply:
Done.